.github/workflows/cd.yml

```diff
@@ -0,0 +1,45 @@
+name: cd-front
+run-name: ${{ github.actor }} is executing cd-front
+
+permissions:
+  id-token: write # This is required for requesting the JWT
+  contents: read  # This is required for actions/checkout
+
+# run on merge since you cannot push to main directly
+on:
+  push: 
+    branches:
+      - main 
+
+defaults:
+  run:
+    working-directory: front
+
+jobs:
+  cd-webapp:
+    runs-on: ubuntu-latest
+    steps:
+      # Followed https://github.com/aws-actions/configure-aws-credentials
+      # Set up GitHub OIDC using https://docs.github.com/en/actions/deployment/security-hardening-your-deployments/configuring-openid-connect-in-amazon-web-services
+      - name: Configure AWS credentials 
+        uses: aws-actions/configure-aws-credentials@v2
+        with:
+          aws-region: eu-west-2
+          role-to-assume: arn:aws:iam::111982095069:role/github-omnilog
+          role-session-name: GitHubActions-CD-Front
+      - name: Checkout main
```

Comment:
is this necessary? I think this happens by default fyi

Reply:
(checking out to main)

Reply:
Without checkout, the workflow fails because the directory doesn't exist:

![image](https://github.com/Theodo-UK/OmniLog/assets/57725347/6fe74f37-5487-4079-af30-531a1747c820)